crates/router_derive/src/macros/operation.rs

```diff
@@ -231,103 +195,279 @@ impl Conversion {
+
+    custom_keyword!(operations);
+    custom_keyword!(flow);
+}
+
+#[derive(Debug)]
+pub enum OperationsEnumMeta {
+    Operations {
+        keyword: operations_keyword::operations,
+        value: Vec<Conversion>,
+    },
+    Flow {
+        keyword: operations_keyword::flow,
+        value: Vec<Derives>,
+    },
+}
+
+#[derive(Clone)]
+pub struct OperationProperties {
+    operations: Vec<Conversion>,
+    flows: Vec<Derives>,
+}
+
+fn get_operation_properties(
+    operation_enums: Vec<OperationsEnumMeta>,
+) -> syn::Result<OperationProperties> {
+    let mut operations = vec![];
+    let mut flows = vec![];
+
+    for operation in operation_enums {
+        match operation {
+            OperationsEnumMeta::Operations { value, .. } => {
+                operations = value;
+            }
+            OperationsEnumMeta::Flow { value, .. } => {
+                flows = value;
+            }
+        }
+    }
+
+    if operations.is_empty() {
+        Err(syn::Error::new(
+            Span::call_site(),
+            "atleast one operation must be specitied",
+        ))?;
+    }
+
+    if flows.is_empty() {
+        Err(syn::Error::new(
+            Span::call_site(),
+            "atleast one flow must be specitied",
+        ))?;
+    }
+
+    Ok(OperationProperties { operations, flows })
+}
+
+impl Parse for Derives {
+    fn parse(input: syn::parse::ParseStream<'_>) -> syn::Result<Self> {
+        let text = input.parse::<syn::LitStr>()?;
+        let value = text.value();
+
+        value.as_str().parse().map_err(|_| {
+            let possible_values = Derives::iter()
+                .map(|variants| variants.to_string())
+                .collect::<Vec<_>>()
+                .join(", ");
+
+            syn::Error::new_spanned(
+                &text,
+                format!(
+                    "Unexpected value for flow: `{value}`. Possible values are `{possible_values}`"
+                ),
+            )
         })
-        .cloned()
-        .ok_or_else(|| {
-            helpers::syn_error(
-                Span::call_site(),
-                "Cannot find attribute 'operation' in the macro",
+    }
+}
+
+impl Parse for Conversion {
+    fn parse(input: syn::parse::ParseStream<'_>) -> syn::Result<Self> {
+        let text = input.parse::<syn::LitStr>()?;
+        let value = text.value();
+
+        value.as_str().parse().map_err(|_| {
+            let possible_values = Conversion::iter()
+                .map(|variants| variants.to_string())
+                .collect::<Vec<_>>()
+                .join(", ");
+
+            syn::Error::new_spanned(
+                &text,
+                format!("Unexpected value for operation: `{value}`. Possible values are `{possible_values}`"),
             )
         })
+    }
+}
+
+fn get_conversions(input: syn::parse::ParseStream<'_>) -> syn::Result<Vec<Conversion>> {
+    let lit_str_list = input.parse::<syn::LitStr>()?;
+    lit_str_list
+        .value()
+        .split(",")
```

Comment:
If it helps for readability, we can allow for optional spaces before and after the comma by trimming each element before parsing it as `Conversion` using `Conversion::from_str`.

This would allow us to write things like:

```rust
#[operation(
    operations = "post_update_tracker",
    flow = "sync_data, authorize_data, cancel_data, capture_data, complete_authorize_data, approve_data, reject_data, setup_mandate_data, session_data"
)]
```

Reply:
Yes, it will for sure be good for readibility